### PR TITLE
ftx: fix null currency.Pair UnmarshalJSON

### PR DESCRIPTION
**Pull request by geseq**
# PR Description

I'm not sure this is the best place for this fix, but with FTX on some submit order calls (spot, with IOC), the returned data includes both `market` with the asset pair and `future` with `null`.

The `null` ends up in currency pair `UnmarshalJSON` resulting in an error for a successful SubmitOrder call.
I am not entirely sure if this will be a breaking change

If this looks ok at face value I'm happy to add some unit tests[

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [x] go test ./... -race
- [x] golangci-lint run
- [x] Manually

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### CODE REVIEW

File: currency/pair_methods.go [hunk 1]
```diff
@@ -34,6 +34,11 @@ func (p *Pair) UnmarshalJSON(d []byte) error {
 		return err
 	}
 
+	if pair == "" {
```
Comment (gloriousCode) [inline currency/pair_methods.go:37]: Please add a test case for `TestPairUnmarshalJSON` with
```go
encoded, err = json.Marshal(EMPTYPAIR)
	err = json.Unmarshal(encoded, &unmarshalHere)
	if err != nil {
		t.Fatal("Pair UnmarshalJSON() error", err)
	}
```

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Hello!
When speaking specifically about the FTX submit order, I made the following changes in another PR #1023 That field was definitely annoying, even with futures orders and that change fixes up the issue. 

image

I also share the concern of it breaking in allowing more currency pair slips to come through, and now needing an extra `IsEmpty()` check. I'll need to do more testing - replacing your `return nil` with `os.Exit(-1)` has not yielded any unexpected exits though. It may be simpler to just change the `currency.Pair` property to `string`

**Review by shazbert — COMMENTED** [review]
Happy with this change. 👍 Agree with the tests then should be good to go.

**Review by gloriousCode — APPROVED** [review]
tACK! The only area that I identified a potential EMPTYPAIR slip-through spot was websocket order processing. However, its still preferable to store and track an order with an empty pair than it would be to error as it unmarshalls the JSON with a nil pair entry.

All other areas of concern for me had some form of checking or handled the situation appropriately. So thank you for your contribution!

File: currency/pair_test.go [hunk 2]
```diff
@@ -70,6 +70,22 @@ func TestPairUnmarshalJSON(t *testing.T) {
...
+	err = json.Unmarshal(encoded, &unmarshalHere)
+	if err != nil {
+		t.Fatal("Pair UnmarshalJSON() error", err)
+	}
+	err = json.Unmarshal([]byte("null"), &unmarshalHere)
+	if err != nil {
+		t.Fatal("Pair UnmarshalJSON() error", err)
+	}
+	if unmarshalHere != EMPTYPAIR {
+		t.Fatalf("Expected EMPTPAIR got: %s", unmarshalHere)
```
Comment (thrasher-) [inline currency/pair_test.go:87]:
```suggestion
		t.Fatalf("Expected EMPTYPAIR got: %s", unmarshalHere)
```
😄

**Review by thrasher- — CHANGES_REQUESTED** [review]
One minor nit and then this looks G2G 👍

**Review by shazbert — APPROVED** [review]
Changes ACK!

**Review by thrasher- — APPROVED** [review]

**Comment by thrasher-** [comment]
Thanks @geseq !
